test/e2e/version_test.go

```diff
@@ -28,10 +28,13 @@ import (
 func TestVersion(t *testing.T) {
 	t.Parallel()
 
-	r := test.NewKnRunResultCollector(t)
+	it, err := test.NewKnTest()
```

Comment:
`NewKnTest()` will create a namespace for this test.
1. We don't need namespace creation for version test.
2. We dont have `tearDown` in this test, which leaving behind the created namespace.
Ran this locally from PR:
```
➜  client git:(c8c54c52) go test ./test/e2e/ -count=1 -test.v --tags e2e -run TestVersion                                                                        
=== RUN   TestVersion
=== PAUSE TestVersion
=== CONT  TestVersion
--- PASS: TestVersion (5.54s)
PASS
ok  	knative.dev/client/test/e2e	5.544s

➜  client git:(c8c54c52) kubectl get ns|grep kne2etests
kne2etests0                                             Active   56s
```
If `kn` doesn't require a namespace to execute, lets not create one? Call to `NewKnTest()` will create one.

I'd still suggest the earlier review comment fix for version test, no need to define a new method for only version test case.

```
func TestVersion(t *testing.T) {
  t.Parallel()
  r := test.NewKnRunResultCollector(t, nil)
  defer r.DumpIfFailed()
  out := test.RunKn("", []string{"version"})
  r.AssertNoError(out)
  assert.Check(t, util.ContainsAll(out.Stdout, "Version"))
}
```

Reply:
If we go without this, we'll have a namespace left behind from e2e test run which will break running e2e locally against same cluster in next run.

Reply:
I think having a way to run w/o namespace is important since remember that this code is now library that can be used outside. I added code to tear down the namespace when the test is done:

```go
+	defer func() {
+		assert.NilError(t, it.Teardown())
+	}()
```

Reply:
Note also that the new code:

```golang
// RunNoNamespace the 'kn' CLI with args but no namespace
func (k Kn) RunNoNamespace(args ...string) KnRunResult {
	return RunKn("", args)
}
```

Is pretty much just a wrapper to make clear we are running `kn` on default namespace which is a useful thing in general. Keeps the `VersionTest` consistent with the other tests. We still have various `e2e` tests that need to be converged to this model so I think it's best to keep the simple `VersionTest` in line with the others since people will use that as an exemplar.

Reply:
Well, as `test.RunKn` is an exported method, this could actually be an example for how to run `kn` without creating a namespace in tests. Say a plugin does not require a namespace to perform its task, its e2e can refer version tests.